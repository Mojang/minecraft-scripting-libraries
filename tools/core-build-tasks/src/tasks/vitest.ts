import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
