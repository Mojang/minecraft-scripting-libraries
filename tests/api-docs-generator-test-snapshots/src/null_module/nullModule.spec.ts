// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Null Module', () => {
    it('Correctly processes null module member categories when changelogged', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['ts', 'msdocs', 'changelog-json'],
        });
    });
});
