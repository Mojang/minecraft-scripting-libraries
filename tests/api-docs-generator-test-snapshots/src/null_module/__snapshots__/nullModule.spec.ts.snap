// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Null Module > Correctly processes null module member categories when changelogged > # Of Files Generated 1`] = `8`;

exports[`Null Module > Correctly processes null module member categories when changelogged > changelog/test-module.json 1`] = `
"[
  {
    "version": "2.0.0",
    "dependencies": [],
    "peer_dependencies": [],
    "classes": [],
    "interfaces": [],
    "errors": [],
    "objects": [],
    "functions": [],
    "constants": [],
    "enums": [],
    "type_aliases": []
  },
  {
    "version": "1.0.0",
    "$added": true,
    "classes": null,
    "interfaces": null,
    "enums": null,
    "errors": null,
    "constants": null,
    "functions": null,
    "objects": null,
    "type_aliases": null,
    "dependencies": null,
    "peer_dependencies": null,
    "minecraft_version": "1.0.0",
    "name": "test-module",
    "uuid": "9f56871a-9f3a-11ec-b909-0242ac120002",
    "module_type": "script"
  }
]"
`;

exports[`Null Module > Correctly processes null module member categories when changelogged > msdocs/priorscriptapi/TOC.yml 1`] = `
"  - name: test-module 1.x.x
    items:
    - name: test-module 1.x.x
      href: test-module-1xx/test-module.md
"
`;

exports[`Null Module > Correctly processes null module member categories when changelogged > msdocs/priorscriptapi/test-module-1xx/test-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: test-module-1xx Module
description: Contents of the test-module module (Version 1.x.x)
---
# \`test-module\` Module (Version 1.x.x)

> [!IMPORTANT]
> This documentation is for an older version of this module. Go to the latest documentation [*here*](../../../scriptapi/test-module/test-module.md).

## Manifest Details
\`\`\`json
{
    "module_name": "test-module",
    "version": "1.0.0"
}
\`\`\`

## Available Versions
- \`1.0.0\`
"
`;

exports[`Null Module > Correctly processes null module member categories when changelogged > msdocs/scriptapi/TOC.yml 1`] = `
"  - name: test-module
    items:
    - name: test-module
      href: test-module/test-module.md
    - name: test-module Changelog
      href: test-module/changelog.md
"
`;

exports[`Null Module > Correctly processes null module member categories when changelogged > msdocs/scriptapi/test-module/changelog.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: test-module Changelog
description: Changelog of the \`test-module\` module
---
# \`test-module\` Changelog

## Version Changes
- [2.0.0](#200)
- [1.0.0](#100)

## 2.0.0
## 1.0.0
#### Added \`test-module\` Module
"
`;

exports[`Null Module > Correctly processes null module member categories when changelogged > msdocs/scriptapi/test-module/test-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: test-module Module
description: Contents of the test-module module
---
# \`test-module\` Module

## [Changelog](changelog.md)

## Manifest Details
\`\`\`json
{
    "module_name": "test-module",
    "version": "2.0.0"
}
\`\`\`
This is version 2.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`2.0.0\`

### Prior Versions

The following API versions are from a major version that is documented here: [*test-module*](../../../priorscriptapi/test-module-1xx/test-module.md):
- \`1.0.0\`

"
`;

exports[`Null Module > Correctly processes null module member categories when changelogged > typescript/test-module@1.0.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "test-module",
 *   "version": "1.0.0"
 * }
 * \`\`\`
 *
 */
"
`;

exports[`Null Module > Correctly processes null module member categories when changelogged > typescript/test-module@2.0.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "test-module",
 *   "version": "2.0.0"
 * }
 * \`\`\`
 *
 */
"
`;
